bears/general/AnnotationBear.py

```diff
@@ -0,0 +1,130 @@
+        :param annot:    A dict containing start of annotation as key and end of
+                         annotation as value.
+        :return:         A set of SourceRanges giving the range of annotation.
+        """
+        text = ''.join(file)
+        found_pos = set()
+        for annot_type in annot:
+            found_pos.update(unescaped_search_in_between(
+                                           annot_type, annot[annot_type], text))
+        if found_pos:
+            found_pos = set(SourceRange.from_absolute_position(
+                              filename,
+                              AbsolutePosition(file, position.begin.range[0]),
+                              AbsolutePosition(file, position.end.range[1] - 1))
+                            for position in found_pos)
+        return found_pos
+
+    @staticmethod
+    def find_singleline_comments(filename, file, comments):
+        """
+        Finds all single-line comments.
+
+        :param filename: Name of the file on which the bear is running.
+        :param file:     Contents of the file in the form of tuple of lines.
+        :param comments: A list containing different types of
+                         single-line comments.
+        :return:         A set of SourceRange objects with start as the
+                         beginning of the comment and end as
+                         the termination of line.
+        """
+        text = ''.join(file)
+        single_comments = []
+        for comment_type in comments:
+            for found in unescaped_search_for(comment_type, text):
+                start = found.start()
+                end = text.find('\n', start)
+                end = len(text) - 1 if end == -1 else end
+                single_comments.append(SourceRange.from_absolute_position(
+                                                  filename,
+                                                  AbsolutePosition(file, start),
+                                                  AbsolutePosition(file, end)))
+        return set(single_comments)
+
+
+def starts_within_ranges(inside_range, outside_ranges):
+    """
+    Finds if a particular range starts inside a collection of given ranges.
+
+    :param outside_ranges: SourceRange identifying range to be searched.
+    :param inside_range:   A tuple SourceRange objects identifying ranges
+                           to be searched in.
+    :return:               True if inside_range is found inside any of the
+                           ranges given by outside_ranges, else
+                           False is returned.
+    """
+    for outside_range in outside_ranges:
+        if inside_range == outside_range:
+            continue
+
+        # Special case of python language.
+        # Where doc strings (""") and strings (") have a similar start.
+        elif inside_range.start == outside_range.start:
+            if inside_range.end < outside_range.end:
+                return True
+
```

Comment:
can we remove this line and 120, IMO it seperates the elif from the if and makes it harder to grasp what belongs to what